### ISSUE TITLE
Different prediction results using vl_simplenn when different number of images passed 

### ISSUE THREAD

**Issue by sunset123** [vlfeat/matconvnet#416]
Hello,
I am evaluating the accuracy of my network using: 'res = vl_simplenn(net, im)'. My images are 210x280 and have only one channel and 2 classes (attached architecture). I am getting different predictions for the same image if I pass one image in im ('im=imdb.images.data(:,:,:,1);') than if I pass 20 images (im=imdb.images.data(:,:,:,1:20);'). This sometimes results in completely opposite predictions. Can you please advice me what the issue?

Thanks

<img width="1050" alt="net" src="https://cloud.githubusercontent.com/assets/17255997/13056161/855c6a74-d40b-11e5-9d9a-422101598c2d.png">

**Comment by rosenfeldamir** [comment]
@sunset123 , which version are you using? This might have to do with batch normalisation, which in previous versions caused test-time results to be dependent on the specific batch. I think there was some discussion here on this previously.

**Comment by sunset123** [comment]
Thanks Amir. I am using v1.18. Bnorm was the cause of the problem. I applied the deploy function and now I get the same results regardless of the batch size. But the predictions is different from training data (basically all images predicted as one class). 

So I tried the other solution suggested on the forum (https://github.com/vlfeat/matconvnet/issues/263) which is to use test mode in vl_simplenn which gave me similar error as training time.
